### PR TITLE
Add-node buttons

### LINKED ISSUE #9219: New "add component" menu item on component
We want a new + menu item on the small menu that sits underneath the menu like so

image

Pressing it opens the component browser connecting a new component to the existing one.

If the visualisation is open the + button needs to move down under the visualisation

image

When the new component is added it should be positioned 
 - vertically in line with where the + button is when it is pressed
 - horizontally left aligned with component it is being added from

I **think** the + should be grey to match the other wheel icons. Not blue like in the screenshots and the existing + button.

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Introduce add-node button below circular menu or open visualization.

https://github.com/enso-org/enso/assets/1047859/aa6cedba-ca7e-44c5-ab27-2f5d5f9421e8

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by kazcw** [comment]
We have decided to go ahead with this implementation; if any changes to the #9219 design are made we can address them then.

File: app/gui2/e2e/selectingNodes.spec.ts [hunk 1]
```diff
@@ -11,9 +11,16 @@ test('Selecting nodes by click', async ({ page }) => {
...
+  const deselectAll = async () => {
+    await page.keyboard.press('Escape')
+    await expect(node1).not.toBeSelected()
+    await expect(node2).not.toBeSelected()
+  }
+
   await locate.graphNodeIcon(node1).click()
   await expect(node1).toBeSelected()
   await expect(node2).not.toBeSelected()
+  await deselectAll()
```
Comment (farmaazon) [inline app/gui2/e2e/selectingNodes.spec.ts]: This line forfeits one of the covered cases: that clicking at non-selected node also deselect previous one.

File: app/gui2/e2e/selectingNodes.spec.ts [hunk 2]
```diff
@@ -23,14 +30,12 @@ test('Selecting nodes by click', async ({ page }) => {
...
   await expect(node2).toBeSelected()
+  await deselectAll()
 
   await locate.graphNodeIcon(node2).click()
   await expect(node1).not.toBeSelected()
   await expect(node2).toBeSelected()
-
-  await page.mouse.click(600, 200)
-  await expect(node1).not.toBeSelected()
-  await expect(node2).not.toBeSelected()
```
Comment (farmaazon) [inline app/gui2/e2e/selectingNodes.spec.ts]: Another case removed: clicking at the background should deselect all nodes. I actually had regressions there: please restore it.

File: app/gui2/e2e/selectingNodes.spec.ts [hunk 3]
```diff
@@ -23,14 +30,12 @@ test('Selecting nodes by click', async ({ page }) => {
   await locate.graphNodeIcon(node1).click({ modifiers: ['Shift'] })
   await expect(node1).toBeSelected()
   await expect(node2).toBeSelected()
+  await deselectAll()
```
Comment (farmaazon) [inline app/gui2/e2e/selectingNodes.spec.ts]: Here also: I want to test that even having two nodes selected, when clicking one, the other is deselected.

File: app/gui2/src/components/GraphEditor.vue [hunk 4]
```diff
@@ -394,13 +394,18 @@ function showComponentBrowser(nodePosition?: Vec2, usage?: Usage) {
   componentBrowserVisible.value = true
 }
 
-function startCreatingNodeFromButton() {
+function addNodeFromSelection() {
   const targetPos =
     placementPositionForSelection() ??
     nonDictatedPlacement(DEFAULT_NODE_SIZE, placementEnvironment.value).position
   showComponentBrowser(targetPos)
 }
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor.vue]: Technically, it is not only about selection, but I should be able to add a node even without any selected nodes (the position should be the scene origin, pushing down if occupied).

**Review by farmaazon — APPROVED** [review]

**Pull request merged**